core/runtime/src/main/java/io/quarkus/runtime/StartupContext.java

```diff
@@ -70,20 +79,17 @@ public boolean isLastValueSet() {
 
     @Override
     public void close() {
-        runAllInReverseOrder(shutdownTasks);
-        shutdownTasks.clear();
-        runAllInReverseOrder(lastShutdownTasks);
-        lastShutdownTasks.clear();
+        runAll(shutdownTasks);
+        runAll(lastShutdownTasks);
     }
 
-    private void runAllInReverseOrder(List<Runnable> tasks) {
-        List<Runnable> toClose = new ArrayList<>(tasks);
-        Collections.reverse(toClose);
-        for (Runnable r : toClose) {
+    private void runAll(Deque<Runnable> tasks) {
```

Comment:
I know that the method is private but it wouldn't hurt if the name reflects the fact that elemens are removed, i.e. something like `runAndRemoveAll()`, `runAllAndClear()`, ... WDYT?